src/components/OSCALProfileCatalogInheritance.js

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import { TreeItem } from "@material-ui/lab";
+import OSCALProfileCatalogInheritanceTree from "./OSCALProfileCatalogInheritanceTree";
+
+const useStyles = makeStyles((theme) => ({
+  inheritance: {
+    marginTop: theme.spacing(2),
+    marginBottom: theme.spacing(2),
+  },
+}));
+
+function generateLabel(title, type) {
+  return type === "profile" ? `Profile: ${title}` : `Catalog: ${title}`;
+}
+
+function createTree(tree, ids) {
+  if (tree.inherited == null) {
+    return null;
+  }
+
+  const children = [];
+
+  tree.inherited.forEach((childProfileOrCatalog) => {
+    ids.push(`${childProfileOrCatalog.uuid} #${ids.length + 1}`);
+    children.push(
+      <TreeItem
+        nodeId={`${childProfileOrCatalog.uuid} #${ids.length}`}
+        key={`${childProfileOrCatalog.uuid} #${ids.length}`}
+        label={generateLabel(
+          childProfileOrCatalog.title,
+          childProfileOrCatalog.type
+        )}
+      >
+        {createTree(childProfileOrCatalog, ids)}
+      </TreeItem>
+    );
+  });
+
+  return children;
+}
+
+export default function OSCALProfileCatalogInheritance(props) {
+  const classes = useStyles();
+  const [validIds, setValidIds] = useState([false, []]);
```

Comment:
Why do we need all of these instances of IDs separately rather than just using the UUID already in the inherited objects?

Reply:
How this is answered will impact what exact changes are made:

Do we want to ensure each nodeId in the tree is unique? We could have the same profile or catalog in different parts of the tree which could cause repeated nodeIds.

Reply:
If we're setting node IDs and keys we do want them to be unique.

Are we sure we need `useState` here?

I'd also recommend changing these to something like `expandedIds` everywhere.

Reply:
We need `useState` so when we set the expanded ids, we re-render the component. If we do not re-render, the tree will not be expanded by default.